app/src/main/java/com/nextcloud/talk/data/user/UsersDao.kt

```diff
@@ -0,0 +1,155 @@
+    abstract fun getActiveUser(): Maybe<UserEntity>
+
+    @Query("SELECT * FROM User where current = 1")
+    abstract fun getActiveUserSynchronously(): UserEntity?
+
+    @Query("SELECT * FROM User WHERE current = 1")
+    abstract fun getActiveUserLiveData(): Single<UserEntity?>
+
+    @Query("DELETE FROM User WHERE id = :id")
+    abstract fun deleteUserWithId(id: Long)
+
+    @Update
+    abstract fun updateUser(user: UserEntity): Int
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    abstract fun saveUser(user: UserEntity): Long
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    abstract fun saveUsers(vararg users: UserEntity): List<Long>
+
+    // get all users not scheduled for deletion
+    @Query("SELECT * FROM User where scheduledForDeletion != 1")
+    abstract fun getUsers(): Single<List<UserEntity>>
+
+    @Query("SELECT * FROM User where id = :id")
+    abstract fun getUserWithId(id: Long): Maybe<UserEntity>
+
+    @Query("SELECT * FROM User where id = :id AND scheduledForDeletion != 1")
+    abstract fun getUserWithIdNotScheduledForDeletion(id: Long): Maybe<UserEntity>
+
+    @Query("SELECT * FROM User where userId = :userId")
+    abstract fun getUserWithUserId(userId: String): Maybe<UserEntity>
+
+    @Query("SELECT * FROM User where userId != :userId")
+    abstract fun getUsersWithoutUserId(userId: Long): Single<List<UserEntity>>
+
+    @Query("SELECT * FROM User where scheduledForDeletion = 1")
+    abstract fun getUsersScheduledForDeletion(): Single<List<UserEntity>>
+
+    @Query("SELECT * FROM User where scheduledForDeletion = 0")
+    abstract fun getUsersNotScheduledForDeletion(): Single<List<UserEntity>>
+
+    @Query("SELECT * FROM User WHERE username = :username AND baseUrl = :server")
+    abstract fun getUserWithUsernameAndServer(username: String, server: String): Maybe<UserEntity>
+
+    @Transaction
+    open suspend fun setUserAsActiveWithId(id: Long): Boolean {
+        val users = getUsers()
+        var result = TRUE
+
+        users.subscribe(object : SingleObserver<List<UserEntity>> {
+            override fun onSuccess(users: List<UserEntity>) {
+                for (user in users) {
+                    // removed from clause: && UserStatus.ACTIVE == user.status
+                    if (user.id != id) {
+                        user.current = TRUE
+                        updateUser(user)
+                    } // removed from clause: && UserStatus.ACTIVE != user.status
+                    else if (user.id == id) {
+                        user.current = TRUE
+                        updateUser(user)
+                    }
+                }
+            }
+
+            override fun onSubscribe(d: Disposable) {
+                // unused atm
+            }
+
+            override fun onError(e: Throwable) {
+                Log.e(TAG, "Error setting user active", e)
+                result = FALSE
+            }
+        })
+
+        return result
```

Comment:
Since RX2 doesn't have a blocking subscribe, how would I do this? `blockingGet()` + catching an Exception for the error scenario and take it from there?

Reply:
Like I did on markUserForDeletion ?

Reply:
Either do everything synchronously, or drop the `suspend` (which you should do anyway), return a `Single<Boolean>` and use `map` to map the subscription to the result.

Reply:
Done https://github.com/nextcloud/talk-android/pull/2154/commits/5aa05a29705f827319f37bae0fad48e7a69a778a